docs/Extensions.md

```diff
@@ -27,6 +27,14 @@ services:
 
 For explanations of these extensions, please refer to the [Podman Documentation](https://docs.podman.io/).
 
+## Secrets
+```yml
+secrets:
+  custom-secret:
+    x-podman.relabel: z
```

Comment:
I think it makes sense to borrow values of the podman mount option (https://docs.podman.io/en/latest/markdown/podman-run.1.html#mount-type-type-type-specific-option). Bind and glob support `relabel` option with `shared` and `private` values possible.